src/Button/ToggleButton/ToggleButton.jsx

```diff
@@ -93,7 +93,8 @@ class ToggleButton extends React.Component {
    * @param {Object} nextProps The received properties.
    */
   componentWillReceiveProps(nextProps) {
-    if (this.state.pressed !== nextProps.pressed) {
+    // Checks if the pressed property has changed before checking its match with the current state 
+    if (this.props.pressed !== nextProps.pressed && this.state.pressed !== nextProps.pressed) {
```

Comment:
The state should always represent the correct status so the previous check should be enough if the state is always set correctly.